### PR TITLE
[pvr] improve value increase calls

### PR DESCRIPTION
**Pull request by AlwinEsch**
This request is from a old rework from me. This change the `++`where possible to the front and use `auto` for loops.

### CODE REVIEW

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by ksooo** [comment]
While you are at it: How about changing the touched for loops to c++11 range based loops, where possible? => http://en.cppreference.com/w/cpp/language/range-for

**Comment by AlwinEsch** [comment]
Hi @ksooo has changed the loops to `auto`

File: xbmc/pvr/addons/PVRClients.cpp [hunk 1]
```diff
@@ -924,7 +921,7 @@ void CPVRClients::ProcessMenuHooks(int iClientID, PVR_MENUHOOK_CAT cat, const CF
       pDialog->SetHeading(CVariant{19196});
 
       PVR_CLIENTMAP_CITR itrClients;
-      for (itrClients = clients.begin(); itrClients != clients.end(); itrClients++)
+      for (itrClients = clients.begin(); itrClients != clients.end(); ++itrClients)
```
Comment (ksooo) [inline xbmc/pvr/addons/PVRClients.cpp]: Any reason for not using range based for loop here?

File: xbmc/pvr/addons/PVRClients.cpp [hunk 2]
```diff
@@ -934,8 +931,8 @@ void CPVRClients::ProcessMenuHooks(int iClientID, PVR_MENUHOOK_CAT cat, const CF
       if (selection >= 0)
       {
         itrClients = clients.begin();
-        for (int i = 0; i < selection; i++)
-          itrClients++;
+        for (int i = 0; i < selection; ++i)
+          ++itrClients;
         iClientID = itrClients->first;
```
Comment (ksooo) [inline xbmc/pvr/addons/PVRClients.cpp]: Couldn't the whole block be replaced by just  <code>iClientID = clients.at(selection).first;</code?>

Comment (AlwinEsch) [inline xbmc/pvr/addons/PVRClients.cpp]: Was also not sure now, but also with C11 not possible. It still use the `key_type` for `at` :(.
Has looked here: http://www.cplusplus.com/reference/map/map/at/

Comment (fritsch) [inline xbmc/pvr/addons/PVRClients.cpp]: If one needs to access the map at an entry determined by a run variable, map is the wrong datatype in the first place.

Comment (ksooo) [inline xbmc/pvr/addons/PVRClients.cpp]: Imo this highly depends on how often one uses <code>at</code> compared to iterators. Here, the ratio is "healthy" and in this case there is no relevant perfomance impact, if you ask me.

Comment (fritsch) [inline xbmc/pvr/addons/PVRClients.cpp]: What about using: http://en.cppreference.com/w/cpp/iterator/advance instead?

File: xbmc/pvr/addons/PVRClients.cpp [hunk 3]
```diff
@@ -923,20 +920,18 @@ void CPVRClients::ProcessMenuHooks(int iClientID, PVR_MENUHOOK_CAT cat, const CF
...
       int selection = pDialog->GetSelectedItem();
       if (selection >= 0)
       {
-        itrClients = clients.begin();
-        for (int i = 0; i < selection; i++)
-          itrClients++;
-        iClientID = itrClients->first;
+        auto client = clients.begin();
+        std::advance(client, selection);
+        iClientID = client->first;
```
Comment (AlwinEsch) [inline xbmc/pvr/addons/PVRClients.cpp]: Hi @ksooo and @fritsch have changed it and learned a new part from C++, thanks a lot! :D

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**